symphony-bdk-spring/symphony-bdk-app-spring-boot-starter/src/main/java/com/symphony/bdk/app/spring/auth/AppAuthController.java

```diff
@@ -0,0 +1,87 @@
+package com.symphony.bdk.app.spring.auth;
+
+import com.symphony.bdk.app.spring.auth.model.AppInfo;
+import com.symphony.bdk.app.spring.auth.model.AppToken;
+import com.symphony.bdk.app.spring.auth.model.JwtInfo;
+import com.symphony.bdk.core.auth.AppAuthSession;
+import com.symphony.bdk.core.auth.ExtensionAppAuthenticator;
+import com.symphony.bdk.core.auth.exception.AuthInitializationException;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.core.auth.jwt.UserClaim;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.Valid;
+
+@Slf4j
+@RestController
+@RequestMapping("/bdk/v1/app/auth")
+public class AppAuthController {
+
+  @Value("${bdk.app.auth.jwt-cookie.enabled:false}")
+  private Boolean jwtCookieEnable;
+
+  @Value("${bdk.app.auth.jwt-cookie.expire-in:86400}")
+  private Integer jwtCookieExpireInSec;
+
+  private final ExtensionAppAuthenticator extensionAppAuthenticator;
+
+  public AppAuthController(ExtensionAppAuthenticator extensionAppAuthenticator) {
+    this.extensionAppAuthenticator = extensionAppAuthenticator;
+  }
+
+  @PostMapping
+  public ResponseEntity<AppAuthSession> authenticate(@Valid @RequestBody AppInfo appInfo)
+      throws AuthUnauthorizedException {
+    log.debug("App auth step 1: Initializing extension app authentication");
+
+    AppAuthSession authSession = extensionAppAuthenticator.authenticateExtensionApp(appInfo.getAppToken());
```

Comment:
AppAuthSession is part of our model, here I would perhaps introduce a DTO for the reponse to decouple both (to avoid returning new information if we evolve AppAuthSession for instance)

Reply:
For instance here we are returning the symphony token which is probably not what we want at all right?

Reply:
the symphony token should be returned because it's needed for the second step. I guess we can use the AppToken (which is used for the Request body in step 2) for the response?

Reply:
In the legacy one it is not returned

Here it does not show either https://developers.symphony.com/extension/docs/application-authentication#section-application-authentication-sequence I think the idea of get the Ts (symphony token) from the frontend directly and then check it matches with the one stored by the app

Otherwise we by pass the part of the circle happing before step 17

@symphony-thibault do you agree?

Reply:
Correct, `Ts` at [step 17](https://developers.symphony.com/extension/docs/application-authentication#section-application-authentication-sequence) is retrieved from the Symphony Frontend, not returned step 11 by App Backend